doc/source/whatsnew/v2.1.0.rst

```diff
@@ -371,6 +371,7 @@ Bug fixes
 
 Categorical
 ^^^^^^^^^^^
+- Bug in :meth:`CategoricalIndex.remove_categories` , with categories it now retains the original order when removing an element from the categories list(:issue:`53935`).
```

Comment:
Just pushed up a commit that implements both suggestions. Thanks for your help!

Reply:
Unfortunately, that implementation you suggested failed the CI tests.  I believe because we were overriding the default argument of `None` with `True` some of the time.  I just uploaded a version using a Python Ternary to get this to a one liner.  Let me know what you think.